### ISSUE TITLE
Minor byref error improvements

### ISSUE THREAD

**Issue by TIHan** [Microsoft/visualfsharp#5916]
There are a few more byref errors that we could improve upon, but for the most part we have solved 98% of them. This isn't urgent, but want to report it so we have a trace of it.

Here they are:
```fsharp
     type StaticTest private () =
         static member Test (x: byref<int>, y: int) = ()
         static member Test2 (x: inref<int>, y: int) = ()

         // This passes because tup becomes 'int ref * int', which is valid and produces valid code.
        // We include this to test current behavior with inference and byrefs.
        static member PositiveTest(tup) =
            StaticTest.Test(tup)

     let test3 () =
        StaticTest.Test2 // is passing, but probably shouldn't be
```
We have this exact code as part of our test suite to test current behavior.

`test3` actually compiles to the correct AST that allows for it to work, so it's hard to catch it in post inference checks. Ideally, we should change what AST it gets turned into in a way that it can be caught in post inference checks.

Basically `test3`'s sig is really `val test3 : unit -> inref<int> * int -> unit` - this shouldn't compile but does. Calling it doesn't produce a runtime error as well which is interesting.

The other one:
```
let testTuple (x: byref<int> * int) = ()

let runTestTuple () =
    let mutable x = 1
    testTuple (&x, 1)
```

This one I'm not that concerned because today this won't compile. However, we don't receive an error on `testTuple (&x, 1)` which we probably should.

**Comment by dsyme** [comment]
This can be closed
